web3.go

```diff
@@ -268,3 +268,55 @@ func WaitForReceipt(ctx context.Context, client Client, hash common.Hash) (*Rece
 		}
 	}
 }
+
+func FindEventById(abi abi.ABI, id common.Hash) *abi.Event {
+	for _, event := range abi.Events {
+		if event.Id() == id {
+			return &event
+		}
+	}
+	return nil
+}
+func getInputs(args abi.Arguments, indexed bool) []abi.Argument {
+	var out []abi.Argument
+	for _, arg := range args {
+		if arg.Indexed == indexed {
+			out = append(out, arg)
+		}
+	}
+	return out
+}
+
+func ParseReceipt(myabi abi.ABI, receipt *Receipt) (map[string]map[string]interface{}, error) {
+	var output map[string]map[string]interface{}
+	output = make(map[string]map[string]interface{})
```

Comment:
Small suggestions: 

- Drop `var output...` and switch to`output := make(map[string]map[string]interface{})`
- Use a `type Event map[string]interface` to make the nested maps more readable